src/Shader.cpp

```diff
@@ -0,0 +1,76 @@
+#include "Shader.h"
+
+#include <string>
+#include <iostream>
+#include <fstream>
+#include <vector>
+#include <algorithm>
+
+Shader::Shader() = default;
+
+std::string readFile(const char *filePath) {
+    std::string content;
+    std::ifstream fileStream(filePath, std::ios::in);
+
+    if (!fileStream.is_open()) {
+        std::cerr << "Could not read file " << filePath << ". File does not exist." << std::endl;
+        return "";
+    }
+
+    std::string line;
+    while (!fileStream.eof()) {
+        std::getline(fileStream, line);
+        content.append(line + "\n");
+    }
+
+    fileStream.close();
+    return content;
+}
+
+GLuint Shader::LoadShader(const char *vertex_path, const char *fragment_path) {
+    GLuint vertShader = glCreateShader(GL_VERTEX_SHADER);
+    GLuint fragShader = glCreateShader(GL_FRAGMENT_SHADER);
+
+    std::string vertShaderStr = readFile(vertex_path);
+    std::string fragShaderStr = readFile(fragment_path);
+    const char *vertShaderSrc = vertShaderStr.c_str();
+    const char *fragShaderSrc = fragShaderStr.c_str();
+
+    GLint result = GL_FALSE;
+    int success;
+
+    glShaderSource(vertShader, 1, &vertShaderSrc, nullptr);
+    glCompileShader(vertShader);
+
+    glGetShaderiv(vertShader, GL_COMPILE_STATUS, &result);
+    GLchar infoLog[512];
+    if (!result) {
+        glGetShaderInfoLog(vertShader, 512, nullptr, infoLog);
+        std::cerr << "ERROR::SHADER::VERTEX::COMPILATION_FAILED\n" << infoLog << std::endl;
+    }
+
```

Comment:
Думаешь сообщение компилятора об ошибке не может быть больше 512 байт?
```suggestion
    GLint result = GL_FALSE;
    GLint infoLogLength;

    glShaderSource(vertShader, 1, &vertShaderSrc, nullptr);
    glCompileShader(vertShader);
    {
        glGetShaderiv(vertShader, GL_COMPILE_STATUS, &result);
        if (result == GL_FALSE) {
            glGetProgramiv(vertShader, GL_INFO_LOG_LENGTH, &infoLogLength);
            std::string infoLog(infoLogLength, 0);
            glGetShaderInfoLog(vertShader, infoLogLength, &infoLogLength, infoLog.data());
            infoLog.resize(infoLogLength);
            std::cerr << "ERROR::SHADER::VERTEX::COMPILATION_FAILED\n" << infoLog << std::endl;
        }
    }

```

Reply:
Зачем продолжать, если шейдер все равно не компилируется? Чтобы потом получить еще сообщение об ошибке при линковке?